File: examples/order/python/src/health/health.py
from fastapi import APIRouter
from pydantic import BaseModel
import structlog

logger = structlog.get_logger(__name__)

router = APIRouter()


class HealthResponse(BaseModel):
    status: str = "ok"


@router.get("/healthz", response_model=HealthResponse)
async def liveness_check():
    """Liveness probe - checks if the service is running."""
    logger.debug("Liveness check requested")
    return HealthResponse(status="ok")


@router.get("/readyz", response_model=HealthResponse) 
async def readiness_check():
    """Readiness probe - checks if the service is ready to accept requests."""
    # In a real application, you might check:
    # - Database connectivity
    # - External service dependencies
    # - Configuration validity
    # etc.
    
    logger.debug("Readiness check requested")
    return HealthResponse(status="ok")

File: examples/order/rust/src/health.rs
use axum::{
    http::StatusCode,
    response::{IntoResponse, Json},
    routing::get,
    Router,
};
use serde_json::json;
use tracing::instrument;

pub fn router() -> Router<crate::AppState> {
    Router::new()
        .route("/healthz", get(liveness))
        .route("/readyz", get(readiness))
}

#[instrument]
async fn liveness() -> impl IntoResponse {
    (StatusCode::OK, Json(json!({"status": "ok"})))
}

#[instrument]
async fn readiness() -> impl IntoResponse {
    // In a real implementation, you would check database connections,
    // external service availability, etc.
    (StatusCode::OK, Json(json!({"status": "ready"})))
}
